File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/location/RCTMGLNativeUserLocationManager.java
package com.mapbox.rctmgl.components.location;

import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.annotations.ReactProp;

import javax.annotation.Nonnull;

public class RCTMGLNativeUserLocationManager extends ViewGroupManager<RCTMGLNativeUserLocation> {
    public static final String REACT_CLASS = "RCTMGLNativeUserLocation";

    @Nonnull
    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @ReactProp(name = "androidRenderMode")
    public void setAndroidRenderMode(com.mapbox.rctmgl.components.location.RCTMGLNativeUserLocation userLocation,
            String mode) {
        if ("compass".equalsIgnoreCase(mode)) {
            userLocation.setRenderMode(RenderMode.COMPASS);
        } else if ("gps".equalsIgnoreCase(mode)) {
            userLocation.setRenderMode(RenderMode.GPS);
        } else {
            userLocation.setRenderMode(RenderMode.NORMAL);
        }
    }

    @ReactProp(name = "nativeBearingImage")
    public void setNativeBearingImage(com.mapbox.rctmgl.components.location.RCTMGLNativeUserLocation userLocation,
            ReadableMap image) {
        userLocation.setNativeBearingImage(image);
    }

    @ReactProp(name = "nativeShadowImage")
    public void setNativeShadowImage(com.mapbox.rctmgl.components.location.RCTMGLNativeUserLocation userLocation,
            ReadableMap image) {
        userLocation.setNativeShadowImage(image);
    }

    @ReactProp(name = "nativeTopImage")
    public void setNativeTopImage(com.mapbox.rctmgl.components.location.RCTMGLNativeUserLocation userLocation,
            ReadableMap image) {
        userLocation.setNativeTopImage(image);
    }

    @Nonnull
    @Override
    protected RCTMGLNativeUserLocation createViewInstance(@Nonnull ThemedReactContext reactContext) {
        return new RCTMGLNativeUserLocation(reactContext);
    }
}

File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/location/RCTMGLNativeUserLocationManager.kt
package com.mapbox.rctmgl.components.location

import com.facebook.react.bridge.ReadableMap
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewGroupManager
import com.facebook.react.uimanager.annotations.ReactProp
import com.mapbox.rctmgl.utils.Logger
import javax.annotation.Nonnull

class RCTMGLNativeUserLocationManager : ViewGroupManager<RCTMGLNativeUserLocation>() {
    @Nonnull
    override fun getName(): String {
        return REACT_CLASS
    }

    @ReactProp(name = "androidRenderMode")
    fun setAndroidRenderMode(userLocation: RCTMGLNativeUserLocation?, mode: String?) {
        Logger.w("RCTMGLNativeUserLocationManager", "setAndroidRenderMode is deprecated in v10");
    }

    @ReactProp(name = "nativeBearingImage")
    fun setNativeBearingImage(userLocation: RCTMGLNativeUserLocation?, image: ReadableMap) {
        userLocation?.setNativeBearingImage(image);
    }

    @ReactProp(name = "nativeShadowImage")
    fun setNativeShadowImage(userLocation: RCTMGLNativeUserLocation?, image: ReadableMap) {
        userLocation?.setNativeShadowImage(image);
    }

    @ReactProp(name = "nativeTopImage")
    fun setNativeTopImage(userLocation: RCTMGLNativeUserLocation?, image: ReadableMap) {
        userLocation?.setNativeTopImage(image);
    }

    @Nonnull
    override fun createViewInstance(@Nonnull reactContext: ThemedReactContext): RCTMGLNativeUserLocation {
        return RCTMGLNativeUserLocation(reactContext)
    }

    companion object {
        const val REACT_CLASS = "RCTMGLNativeUserLocation"
    }
}